source/language/classical.rst

```diff
@@ -56,6 +59,17 @@ right circular shift, ``rotl`` and ``rotr``, respectively.
    a | b; // Produces "11111111"
    a & b; // Produces "00000000"
 
+For ``uint`` and ``angle``, the results of these operations are defined as if
+the operations were applied to their defined bit representations::
+
+   angle[4] a = 9 * (pi / 8);  // "1001"
+   a << 2;  // Produces pi/2, which is "0100"
+   a >> 2;  // Produces pi/4, which is "0010"
```

Comment:
**suggestion**: Might be clearer to include an assignment e.g.
```c
angle[4] b = a << 2;  // Produces pi/2, which is "0100"
angle[4] c = a >> 2;  // Produces pi/4, which is "0010"
```

Reply:
Would me putting the exact type (`angle[4]`) be enough for you?  I'd wanted to avoid explicitly writing an assignment, because that introduces the possibility of an explicit cast, whereas here I wanted to be super clear we're talking about the type and value of the expression.

Reply:
> angle[4] b = a << 2;  // Produces pi/2, which is "0100"
> angle[4] c = a >> 2;  // Produces pi/4, which is "0010"

Left shift and right shift are the same exact arithmetic operation - in this case, division?
Left shifting an angle by 2 yields `pi` / 2, but right shifting an angle by 2 yields `pi` / 4?

Reply:
Yeah exactly, the intent is that bit-shifting is defined the same way as it is for a `uint` which has the same bit pattern as `angle` (whether simulated or direct).  So bit-shift right by `n` is the same as integer division by `2 ** n`, and bit-shift left by `n` is the same as integer division by `2 ** (-n)`.

(Just for ease of discussion, since the value of `a` has disappeared from the context on GitHub: it's defined as `angle[4] a = 9 * (pi / 8);` which has the effective bit pattern `"1001"`.)

Reply:
> Would me putting the exact type (angle[4]) be enough for you? I'd wanted to avoid explicitly writing an assignment, because that introduces the possibility of an explicit cast, whereas here I wanted to be super clear we're talking about the type and value of the expression.

Ah got it, I think it's fine as it is then.